bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnectionConfig.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.mqtt;
+
+import java.math.BigDecimal;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Contains configuration for a MqttBrokerConnection. Necessary to add a new broker connection the {@link MqttService}.
+ *
+ * @author David Graeff - Initial contribution and API
+ */
+public class MqttBrokerConnectionConfig {
+    // Optional connection name
+    public @Nullable String name;
+    // Connection parameters (host+port+secure)
+    public @Nullable String host;
+    public @Nullable Integer port;
+    public @Nullable Boolean secure = true;
+    // Authentication parameters
+    public @Nullable String username;
+    public @Nullable String password;
+    public @Nullable String clientID;
+    // MQTT parameters
+    public BigDecimal qos = BigDecimal.valueOf(MqttBrokerConnection.DEFAULT_QOS);
+    public Boolean retainMessages = false;
+    public @Nullable String lastWill;
+    /** Keepalive in seconds */
+    public @Nullable BigDecimal keepAlive;
+
+    /**
+     * Return the brokerID of this connection. This is either the name or host:port(:s).
+     * This method might return null, if none of the parameters is set.
```

Comment:
The method does actually _not_ return null but the empty String is nothing is configured.